lib/Sandbox.js

```diff
@@ -59,6 +59,14 @@ export type EnvironmentVarExport = {
   __BUILT_IN_DO_NOT_USE_OR_YOU_WILL_BE_PIPd?: boolean;
 };
 
+export type EsyConfig = {
+  build: ?string;
+  buildOutOfSource: boolean;
```

Comment:
New `"esy.buildOutOfSource" key here. Indicates if a package can build itself out of source directory. If key isn't present — then it is assumed to be `true`. If key is present and set to `false` then we copy `$cur__root` into `$cur__target_dir` before the build.